docs: update getting-started-splunk-setup.md (#2417)
splunk/splunk-connect-for-syslog: docs/gettingstarted/getting-started-splunk-setup.md

  Note that an external load balancer simplifies long-term
maintenance, but if a load balancer is not
available, native load balancing can be used with ten or fewer indexers where HEC is used exclusively for syslog.

  SC4S traffic must be sent to HEC endpoints configured directly on the indexers rather than
an intermediate tier of heavy forwarders.  

2. Create a HEC token to be used by SC4S. Make sure the token has permissions to add events to 'main', '_metrics', and all
event destinations indexes.

3. You can leave "Selected Indexes" blank on the token configuration page so that the token has access to
all indexes, including the `lastChanceIndex`.  If this list is populated, extreme care must be taken to keep it up to date, as an attempt to
send data to an index that is not in this list results in a `400` error from the HEC endpoint and the `lastChanceIndex` will not be
consulted if the index specified in the event is not configured on Splunk.  Note that one bad message can prevent and entire batch from being sent to Splunk.

> 4. If you are not using TLS on SC4S, turn off SSL on global settings for HEC in Splunk. See [Splunk Cloud](http://docs.splunk.com/Documentation/Splunk/7.3.1/Data/UsetheHTTPEventCollector#Configure_HTTP_Event_Collector_on_managed_Splunk_Cloud)

Reviewer: Can we use latest in place of version 7.3.1

> or [Splunk Enterprise](http://dev.splunk.com/view/event-collector/SP-CAAAE6Q) for specific HEC configuration instructions based on your

Reviewer: is this valid url now ?

Author: It seems to work for me, unless you want me to direct it to a different location?

# Splunk setup

> To set up syslog processing with SC4S, perform the following tasks in your Splunk instance:

Reviewer 2: It's a tricky one: SC4S will process syslog regardless Splunk. But if Splunk is not correctly set up, output from SC4S won't be correctly indexed by Splunk. Can me make it more like: "to ensure proper integration/collaboration/setup of SC4S and Splunk" kind of thing?

Reviewer 2: @jenworthington please check this comment

Author: How about "To ensure proper integration for SC4S and Splunk, perform the following tasks in your Splunk instance:"

Reviewer 2: sounds great thank you!

> 1. Create indexes within Splunk.

Author: I added a new line before the steps. I don't see it rendering though, am i doing it wrong?

> 1. Create indexes within Splunk.

Reviewer 2: rendering works great now:

![image](https://github.com/splunk/splunk-connect-for-syslog/assets/139441697/01d0b0d2-6d62-47e8-a377-afe50f48ca65)

https://splunk.github.io/splunk-connect-for-syslog/2417/gettingstarted/getting-started-splunk-setup/

2. Configure your HTTP event collector.

> 3. Create a load balancing mechanism.

Reviewer 2: "Create a load balancing mechanism" is a subsection to "Configure your HTTP event collector"

Reviewer 2: @jenworthington I think it's better to remove point 3 as a subsection of point 2

Author: done

## Step 1: Create indexes within Splunk

> SC4S maps each sourcetype to the following indexes by default:

Reviewer 2: Customers report problems to us because sometimes they don't create those indexes in Splunk, so better to say in the docs that the SC4S's default set of indexes must be created in Splunk

Author: that makes so much more sense!

Reviewer 2: right, but the call for action for the reader is that they must create those indexes or they will have problems, please compare with the original document

Reviewer 2: can we make it explicit:

```suggestion
SC4S maps each sourcetype to the following indexes by default. Make sure to create them in Splunk.
```

* `email`
* `epav`

> * `epintel`

Reviewer 2: @jenworthington can you please add these two indexes too?

> * `epintel`

Author: done

* `infraops`
* `netauth`
* `netdlp`
* `netdns`
* `netfw`
* `netids`
* `netlb`
* `netops`
* `netwaf`
* `netproxy`
* `netipam`
* `oswin`
* `oswinsec`
* `osnix`
* `print`
* `_metrics` (Optional opt-in for SC4S operational metrics; ensure this is created as a metrics index)

> You can also you create your own indexes in Splunk. See [Create custom indexes]( https://docs.splunk.com/Documentation/Splunk/9.2.1/Indexer/Setupmultipleindexes) for more information.

Reviewer 2: can we say that if you use your own indexes in SC4S you also have to create them in Splunk?

## Step 2: Configure your HTTP event collector

See [Use the HTTP event collector](https://docs.splunk.com/Documentation/Splunk/9.2.1/Data/UsetheHTTPEventCollector) for HEC configuration instructions based on your
Splunk type.

Keep in mind the following best practices specific to HEC for SC4S:
* Make sure that the HEC token created for SC4S has permissions to add events to `main`, `_metrics`, and all other event destination indexes.
* You can leave "Selected Indexes" blank on the token configuration page so that the token has access to
all indexes, including the `lastChanceIndex`.  If you do populate this field, take extreme care to keep it up to date; an attempt to
send data to an index that is not in this list results in a `400` error from the HEC endpoint. The `lastChanceIndex` will not be
consulted if the index specified in the event is not configured on Splunk and the entire batch is then not sent to Splunk.
* If you are not using TLS on SC4S, turn off SSL on global settings for HEC in Splunk.

> * SC4S traffic must be sent to HEC endpoints that are configured directly on the indexers.  

Reviewer 2: can we restore the alternative and change it to `should`? `SC4S traffic should be sent to HEC endpoints configured directly on the indexers rather than an intermediate tier of heavy forwarders.`

> ### Step 2: Create a load balancing mechanism

Reviewer 2: `Create a load balancing mechanism` is a subsection of `Configure your HTTP event collector`

Author: i removed it as a numbered step and made it a subtopic instead.

> Create a load balancing mechanism between SC4S and Splunk indexers. See [Set up load balancing](https://docs.splunk.com/Documentation/Splunk/9.2.1/Forwarding/Setuploadbalancingd) for more information. Note that this should not be confused with load balancing between [sources and SC4S](../lb.md). 

Reviewer 2: splunk docs on load balancing don't apply to this case. Can we make it more in the style of: 

```
In some situations, it is necessary to ensure balancing of the output from SC4S to Splunk indexers. Note that this should not be confused with load balancing between [sources and SC4S](../lb.md). 
```

what we mean here is that:
- if you have Splunk Cloud no worry, you're already covered and your SC4S output will be automatically load balanced to Splunk indexers
- if you have Splunk Enterprise and a single indexer, you obviously don't need an lb
- if you have Splunk Enterprise and mutliple indexers, you should load balance your SC4S output

Author: thanks! Is there another topic we should link to in case they are now to the product and need some guidance for creating this type of load balancing?

Reviewer 2: Unfortunately there's nothing to link at this point, we don't provide any further recommendations for lbs at this point

Author: "In some configurations, you should ensure output balancing from SC4S to Splunk indexers. To do this, you create a load balancing mechanism between SC4S and Splunk indexers. See [Set up load balancing](https://docs.splunk.com/Documentation/Splunk/9.2.1/Forwarding/Setuploadbalancingd) for more information."

When configuring your load balancing mechanism, Keep in mind the following:

* Splunk Cloud provides an internal ELB on TCP 443.
* For Splunk Enterprise set up your Splunk HTTP Event Collector with the HEC endpoints behind a load balancer. 
* An external load balancer simplifies long-term maintenance by eliminating the need to manually keep the list of HEC URLs specified in SC4S current. Set up a load balancer using virtual IP and configured for https round-robin without sticky session. 
* If a load balancer is not available, you can configure a list of HEC endpoint URLs with native syslog-ng load balancing. For internal load balancing of syslog-ng you should:
    * Load balance ten or fewer indexers.

>     * Bse HEC exclusively for syslog.

Reviewer 2: ```suggestion
    * Use HEC exclusively for syslog.
```

# Splunk setup
To ensure proper integration for SC4S and Splunk, perform the following tasks in your Splunk instance:

1. Create copies of your SC4S indexes in Splunk.
2. Configure your HTTP event collector.

## Step 1: Create indexes within Splunk

SC4S maps each sourcetype to the following indexes by default:

* `email`
* `epav`
* `epintel`
* `fireeye`
* `gitops`
* `infraops`
* `netauth`
* `netdlp`
* `netdns`
* `netfw`
* `netids`
* `netlb`
* `netops`
* `netwaf`
* `netproxy`
* `netipam`
* `oswin`
* `oswinsec`
* `osnix`
* `print`
* `_metrics` (Optional opt-in for SC4S operational metrics; ensure this is created as a metrics index)

If you create custom indexes in SC4S you must also create them in Splunk. See [Create custom indexes]( https://docs.splunk.com/Documentation/Splunk/9.2.1/Indexer/Setupmultipleindexes) for more information.

## Step 2: Configure your HTTP event collector

See [Use the HTTP event collector](https://docs.splunk.com/Documentation/Splunk/9.2.1/Data/UsetheHTTPEventCollector) for HEC configuration instructions based on your
Splunk type.

Keep in mind the following best practices specific to HEC for SC4S:

> * Make sure that the HEC token created for SC4S has permissions to add events to `main`, `_metrics`, and all other event destination indexes.

Reviewer 2: `_metrics` are not events, so maybe:

```suggestion
* Make sure that the HEC token created for SC4S has permissions to write to `_metrics` and all event destination indexes.
```
